include/uuid/uuid.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _UUID_H_
+#define _UUID_H_
+
+#include <algorithm>
+#include <bitset>
+#include <chrono>
+#include <cstddef>
+#include <ios>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <sstream>
+#include <string>
+
+#include "random_gen.h"
+
+
+namespace uprotocol::uuid {
```

Comment:
You need to comment your code so people know what this is doing. It seems you are only supporting uuidv8, then the factory should state that it is a UUIDv8Factory but the datamodel can still be the official UUID (mentioned lower in this document). You need to handle if someone passes you a v4, v6, etc.. to handle and report variant, etc...

Reply:
Hi @stevenhartley 
How is someone going to pass us v4/v6 UUID`s as we only support v8 in C++ ? there is no way to generate it.

Also - how can we know what is the version of the UUID?